dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java

```diff
@@ -205,79 +201,19 @@ protected boolean changeIfDockerfileBaseImageLine(String imageToFind, String tag
-        } catch (IOException e) {
-            storeRepo = gitHubUtil.createPublicRepo(store);
-        }
-        updateStoreOnGithub(storeRepo, Constants.STORE_JSON_FILE, img, tag);
-    }
-
-    protected void updateStoreOnGithub(GHRepository repo, String path, String img, String tag) throws IOException {
-        try {
-            repo.getFileContent(path);
-        } catch (IOException e) {
-            repo.createContent("", "initializing store", path);
-        }
-
-        String latestCommit = repo.getBranches().get(repo.getDefaultBranch()).getSHA1();
-        log.info("Loading image store at commit {}", latestCommit);
-        GHContent content = repo.getFileContent(path, latestCommit);
-
-        if (content.isFile()) {
-            JsonElement json;
-            try (InputStream stream = content.read(); InputStreamReader streamR = new InputStreamReader(stream)) {
-                try {
-                    json = new JsonParser().parse(streamR);
-                } catch (JsonParseException e) {
-                    log.warn("Not a JSON format store. Clearing and rewriting as JSON...");
-                    json = JsonNull.INSTANCE;
-                }
-            }
-            String jsonOutput = getAndModifyJsonString(json, img, tag);
-            content.update(jsonOutput,
-                    String.format("Updated image %s with tag %s.\n@rev none@", img, tag), repo.getDefaultBranch());
-        }
-    }
-
-    protected String getAndModifyJsonString(JsonElement json, String img, String tag) throws IOException {
-        JsonElement images;
-        if (json.isJsonNull()) {
-            json = new JsonObject();
-            images = new JsonObject();
-            json.getAsJsonObject().add("images", images);
-        }
-        images = json.getAsJsonObject().get("images");
-        if (images == null) {
-            images = new JsonObject();
-            json.getAsJsonObject().add("images", images);
-            images = json.getAsJsonObject().get("images");
-        }
-        JsonElement newTag = new JsonPrimitive(tag);
-        images.getAsJsonObject().add(img, newTag);
-
-        Gson gson = new GsonBuilder().setPrettyPrinting().create();
-        return gson.toJson(json);
+    public GitHubJsonStore getGitHubJsonStore(String store) {
+        return new GitHubJsonStore(this.gitHubUtil, store);
     }
 
     public void createPullReq(GHRepository origRepo,
                               String branch, GHRepository forkRepo,
-                              String message) throws InterruptedException, IOException {
-        if (message == null) {
-            message = "Automatic Dockerfile Image Updater";
+                              String title) throws InterruptedException, IOException {
+        if (title == null) {
+            title = "Automatic Dockerfile Image Updater";
         }
+        // TODO: This may loop forever in the event of constant -1 pullRequestExitCodes...
```

Comment:
Give it a configurable timeout?

Reply:
Yeah, right now focusing on solidifying the move to stop forking and branch instead. This needs to be done at some point or another though.